linux_initramfs/pack.sh

```diff
@@ -24,47 +24,108 @@
 #cd -
 
 #cp  ../agents/linux_x86_64/bin/loader/loader rootTemplate/loader
-cd ../packer/linux_x86_64-userspace/
+cd ../packer/linux_x86_64-userspace/ || exit
 sh compile_loader.sh
-cd -
+cd - || exit
 cp ../packer/linux_x86_64-userspace/bin64/loader rootTemplate/loader
+
+# copy all lib listed in `lld $arg1` to dir `rootTemplate`
+copy_lib_by_prog(){
+	prog_name=$1
+	# grep: get dyn dependency instead of static lib
+	# sed: remove string's left part splited by "=>"
+	# awk: get the first part of output(lib's absolute path)
+	lib_paths=$(ldd $prog_name | grep "=>" | sed -E "s/^.*=> //" | awk '{print $1}')
+
+	for lib_path in $lib_paths; do
+	target_path="rootTemplate$lib_path"
+	target_dir=$(dirname "$target_path")
+	if [ ! -d "$target_dir" ] ; then
+		mkdir -pv "$target_dir"
+	fi
+	if [ ! -c "$target_path" ] ; then
+		cp -L "$lib_path" "$target_path"
+	fi	
+	done
+}
+
+copy_lib_by_name(){
+	lib_name=$1
+	# grep: get dependency of lib; sed: remove string's left part splited by "=>"
+	lib_paths=$(ldconfig -p | grep -E  "$lib_name \(.*\) => " | sed -E "s/^.*=> //")
+	if [ -z "$lib_paths" ]; then
+		echo "lib $lib_name not found in ldconfig, consider install it"
+		exit 1
+	fi
+
+	for lib_path in $lib_paths; do
+	target_path="rootTemplate$lib_path"
+	target_dir=$(dirname "$target_path")
+	if [ ! -d "$target_dir" ] ; then
+		mkdir -pv "$target_dir"
+	fi
+		cp -Lv "$lib_path" "$target_path"
+	done
+
+}
 #cp /home/kafl/nyx_fuzzer_snapshot/snapshot_toy_examples/packer/linux_x86_64-userspace/bin64/loader rootTemplate/loader
 chmod +x rootTemplate/loader
-mkdir rootTemplate/lib/
-mkdir rootTemplate/lib64/
+mkdir rootTemplate/usr/lib/
+mkdir rootTemplate/usr/lib64/
+mkdir rootTemplate/usr/lib32/
 mkdir rootTemplate/lib/i386-linux-gnu/
```

Comment:
These mkdir's fail with `mkdir: cannot create directory ‘rootTemplate/lib/i386-linux-gnu/’: No such file or directory` so `mkdir -p` would probably be a good idea as well.